### DISCUSSION TITLE
Multiple queries

### DISCUSSION THREAD

**Discussion by demeaux** [DaxStudio/DaxStudio#793] [General]
I've only just started using DAX Studio and so far so good! 
I have to connect to a tabular server which holds multiple databases with multiple tables, I would like to access 15 tables (or thereabouts) from different databases. Is it possible to:

1) write a single query to say 
EVALUATE (db1[table12])
Write to this location
EVALUATE (db2[table1])
Write to this location

or

2) build all 12 queries separately and then have them run as a batch query

I can't seem to find this on https://daxstudio.org/ but I may have missed something.

Thanks.

**Comment by dgosbell** [comment]
There is currently no way to run queries against multiple databases in a single batch using DAX Studio. The way Microsoft built the DAX query language does not support querying objects outside of the database specified in the current connection. Although I actually have some ideas for enabling something like the first option which I hope to implement later this year.

I'm not really sure I understand option 2 are you maybe talking about executing 12 separate queries using a command line interface? If so that is not currently supported by DAX Studio, but it is something I would like to add soon.

One option to do this at the moment would be to use the SQL Server powershell cmdlets to execute your queries. The [Invoke-AsCmd](https://docs.microsoft.com/en-us/powershell/module/sqlserver/invoke-ascmd?view=sqlserver-ps) cmdlet would enable you to write a script to execute your 12 queries in a single script.

**Comment by demeaux** [comment]
> I'm not really sure I understand option 2 are you maybe talking about executing 12 separate queries using a command line interface? If so that is not currently supported by DAX Studio, but it is something I would like to add soon.

that's exactly what I mean!  :)

> One option to do this at the moment would be to use the SQL Server powershell cmdlets to execute your queries. The Invoke-AsCmd cmdlet would enable you to write a script to execute your 12 queries in a single script.

Ok, that is really helpful, I will look into this in the next day or two. 

Thanks for a really rapid reply! (and all the hard work with DAX studio)
